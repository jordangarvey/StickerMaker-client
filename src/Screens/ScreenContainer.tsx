import React, { Component, SFC } from "react";
import { ScrollView, StyleSheet } from "react-native";

import * as C from "../Global/Colours";


/**
 * Component to render the screen container
 */
const ScreenContainer: SFC<Component> = (props) => (
	<ScrollView
		alwaysBounceVertical={ false }
		indicatorStyle="black"
		keyboardDismissMode="interactive"
		style={styles.screen}
	>
		{props.children}
	</ScrollView>
);

/**
 * Styles for the login screen
 */
const styles = StyleSheet.create({
	screen: {
		...StyleSheet.absoluteFillObject,
		backgroundColor: C.background,
		paddingTop: 50,
		paddingHorizontal: 40
	}
});

export default ScreenContainer;
