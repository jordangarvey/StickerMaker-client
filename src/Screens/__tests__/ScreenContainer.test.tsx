import React from "react";
import { Text } from "react-native";
import renderer from "react-test-renderer";

import ScreenContainer from "../ScreenContainer";


describe("The Screen Container componentâ€™s snapshot", () => {
	it("renders as expected", () => {
		const component = renderer.create(
			<ScreenContainer
				children={<Text>Test</Text>}
			/>
		).toJSON();


		expect(component).toMatchSnapshot();
	});
});
