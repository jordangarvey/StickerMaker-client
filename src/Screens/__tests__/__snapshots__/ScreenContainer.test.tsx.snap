// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Screen Container componentâ€™s snapshot renders as expected 1`] = `
<RCTScrollView
  alwaysBounceVertical={false}
  indicatorStyle="black"
  keyboardDismissMode="interactive"
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "bottom": 0,
      "left": 0,
      "paddingHorizontal": 40,
      "paddingTop": 50,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View>
    <Text>
      Test
    </Text>
  </View>
</RCTScrollView>
`;
